### PR TITLE
Send and Receive success screens

### PR DESCRIPTION
**Pull request by benalleng**
Adds new success Screens for Send and Receive as well as making some adjustments to the functionality of the Amount.tsx component to better handle its usecases

### WONTFIX:

- only adds an individual `<AmountFiat />` component for use in the swap route and does not separate all instances of `<Amount />` into its two parts as we are intending to do later on
- error hints or error help needs to be added for each error code sent from mutiny-node
- confirmation time estimates need to be added as functionality from mutiny-node
- transaction fee help copy still needs to be send-type-correct in other words all transaction fee help returns the same message at the moment
- transaction details screens need to be implemented, unless the link directs to a mempool.space transaction all view transaction links are hidden

### CODE REVIEW

File: src/routes/Receive.tsx [hunk 1]
```diff
@@ -365,43 +347,38 @@ export default function Receive() {
                         </Match>
                         <Match when={unified() && receiveState() === "show"}>
                             <FeeWarning fee={lspFee()} flavor={flavor()} />
-                            <IntegratedQr
```
Comment (futurepaul) [inline src/routes/Receive.tsx:368]: Not sure how this happened but you seem to be reverting some work here. (the new QR design)

File: src/routes/Send.tsx [hunk 2]
```diff
@@ -131,10 +132,9 @@ function DestinationInput(props: {
             <SmallHeader>Destination</SmallHeader>
             <textarea
                 value={props.fieldDestination}
-                onInput={(e) => {
-                    const trim = e.currentTarget.value.trim();
```
Comment (futurepaul) [inline src/routes/Send.tsx:135]: revert

File: src/routes/Send.tsx [hunk 3]
```diff
@@ -385,9 +385,8 @@ export default function Send() {
 
         try {
             const text = await navigator.clipboard.readText();
-            const trimText = text.trim();
```
Comment (futurepaul) [inline src/routes/Send.tsx:388]: revert

**Review by futurepaul — CHANGES_REQUESTED** [review]
not sure how you convinced git to go along with it but somehow you're overwriting some recent commits... not sure how widespread it is but these are a few that stood out to me

**Comment by benalleng** [comment]
> somehow you're overwriting some recent commits

Hmm ok... I'll try to resolve that

File: src/components/successfail/MegaCheck.tsx
Comment (benalleng) [inline src/components/successfail/MegaCheck.tsx:1]: Decided not to keep the size changes from the fixed classes when importing this svg directly.
Below is what I had as of [8285876](https://github.com/MutinyWallet/mutiny-web/tree/8285876cfe2e3004f5022382f0de61b4304a9b22)

`mx-auto w-[144px] h-[144px]`

same goes for `<MegaEx/>`

File: src/routes/Receive.tsx [hunk 4]
```diff
@@ -412,45 +406,82 @@ export default function Receive() {
...
+                                        ? "Payment Received"
+                                        : "Payment Initiated"}
+                                </LargeHeader>
                                 <Amount
-                                    amountSats={paymentInvoice()?.amount_sats}
+                                    amountSats={
+                                        receiveState() === "paid" &&
+                                        paidState() === "lightning_paid"
+                                            ? paymentInvoice()?.amount_sats
+                                            : paymentTx()?.received
```
Comment (benalleng) [inline src/routes/Receive.tsx:420]: Would you prefer that the logic behind this be handled more similarly to `Send.tsx`. In other words the `sentDetails()?.amount` looks to see the transaction type ahead of time instead of using a ternary like it is here.

**Review by futurepaul — CHANGES_REQUESTED** [review]

File: src/routes/Send.tsx [hunk 5]
```diff
@@ -390,6 +392,7 @@ export default function Send() {
             const trimText = text.trim();
             setFieldDestination(trimText);
             parsePaste(trimText);
+            parsePaste(trimText);
```
Comment (benalleng) [inline src/routes/Send.tsx:395]: doubled

File: src/components/Amount.tsx [hunk 6]
```diff
@@ -39,28 +43,79 @@ export function Amount(props: {
...
+                            "text-sm": props.size === "small",
+                            "text-lg": props.size === "xl"
+                        }}
+                    >
+                        <Show
+                            when={
+                                props.amountSats && Number(props.amountSats) > 1
+                            }
+                        >
+                            {props.size === "xl" ? "SATS" : "sats"}
```
Comment (futurepaul) [inline src/components/Amount.tsx:77]: I think we should just be consistent with all-caps SATS

File: src/routes/Swap.tsx [hunk 7]
```diff
@@ -283,36 +282,38 @@ export default function Swap() {
...
-                                        ""
-                                    }
+                            <Match when={channelOpenResult()?.channel}>
+                                <MegaCheck />
+                                <div class="flex flex-col justify-center">
+                                <LargeHeader centered>
+                                    Swap Initiated
+                                </LargeHeader>
+                                <p class="text-2xl text-center">
+                                    +{channelOpenResult()?.channel?.balance.toString() ?? "0"} sats will be added to your Lightning balance
```
Comment (futurepaul) [inline src/routes/Swap.tsx:309]: need to format this string

Screenshot 2023-07-19 at 3 18 00 PM

File: src/routes/Send.tsx [hunk 8]
```diff
@@ -577,26 +579,55 @@ export default function Send() {
...
+                                        Transaction Fees
+                                    </p>
+                                    <div class="flex items-start gap-1">
+                                        <Amount
+                                            amountSats={
+                                                sentDetails()?.fee_estimate
+                                            }
+                                            align="right"
+                                            size="small"
+                                            showFiat
```
Comment (futurepaul) [inline src/routes/Send.tsx:624]: size small align right:

small USD is missing a space
it should be 0 SATS, not just 0, right?

Screenshot 2023-07-19 at 3 34 22 PM

File: src/routes/Receive.tsx [hunk 9]
```diff
@@ -466,45 +460,84 @@ export default function Receive() {
...
+                                            />
+                                            <div class="flex items-start py-[1px]">
+                                                <FeesModal icon />
+                                            </div>
+                                        </div>
+                                    </div>
+                                </Show>
+                                {/*TODO: Confirmation time estimate still not possible needs to be implemented in mutiny-node first
+                                    
+                                    <Show
```
Comment (futurepaul) [inline src/routes/Receive.tsx:507]: can delete the jsx here, just leave the comment

File: src/components/layout/Button.tsx [hunk 10]
```diff
@@ -12,9 +12,9 @@ const button = cva(
             intent: {
                 active: "bg-white text-black border border-white hover:text-[#3B6CCC]",
                 inactive:
-                    "bg-black text-white border border-white hover:text-[#3B6CCC]",
+                    "bg-transparent text-white border border-white hover:text-[#3B6CCC]",
                 glowy: "bg-black/10 shadow-xl text-white border border-m-blue hover:m-blue-dark hover:text-m-blue",
-                blue: "bg-m-blue text-white shadow-inner-button hover:bg-m-blue-dark text-shadow-button",
+                blue: "bg-m-blue-400 text-white shadow-subtle-bevel hover:bg-m-blue-dark text-shadow-button",
```
Comment (futurepaul) [inline src/components/layout/Button.tsx:17]: why did you change this button?

File: src/components/MoreInfoModal.tsx [hunk 11]
```diff
@@ -1,20 +1,30 @@
...
-                {i18n.t("more_info_modal_p2")}
-            </p>
+        <MoreInfoModal
+            title={i18n.t("whats_with_the_fees")}
+            linkText={
+                props.icon ? (
+                    <img
+                        src={help}
+                        alt="help"
+                        class="w-[16px] h-[16px] cursor-pointer"
```
Comment (futurepaul) [inline src/components/MoreInfoModal.tsx:19]: use tailwind units when possible. `w-4 h-4`

**Review by futurepaul — APPROVED** [review]
epic, thank you!

**Comment by futurepaul** [comment]
this looks good to me. would appreciate a sanity check from @benthecarman or @TonyGiorgio because both ben and I are a little lost in the sauce at this point

basically: send and receive success / fail screens should look good with roughly the same info they currently hold with a new aesthetic.
